cmd/utils/flags.go

```diff
@@ -1310,12 +1315,85 @@ func SetPlsConfig(ctx *cli.Context, stack *node.Node, cfg *pls.Config) {
 
 	cfg.RootChainURL = ctx.GlobalString(PlasmaRootChainUrlFlag.Name)
 
-	if !ctx.GlobalIsSet(PlasmaRootChainContractFlag.Name) {
-		Fatalf("RootChain contract address must be set, using --rootchain.contract")
+	if ctx.GlobalIsSet(PlasmaRootChainContractFlag.Name) {
+		cfg.RootChainContract = common.HexToAddress(ctx.GlobalString(PlasmaRootChainContractFlag.Name))
 	}
-	cfg.RootChainContract = common.HexToAddress(ctx.GlobalString(PlasmaRootChainContractFlag.Name))
 
-	cfg.Genesis = core.DefaultGenesisBlock(cfg.RootChainContract)
+	switch {
+	case ctx.GlobalBool(TestnetFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 3
+		}
+		cfg.Genesis = core.DefaultTestnetGenesisBlock()
+	case ctx.GlobalBool(RinkebyFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 4
+		}
+		cfg.Genesis = core.DefaultRinkebyGenesisBlock()
+	case ctx.GlobalBool(DeveloperFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 1337
+		}
+		// hard coding for dev mode
+		var (
+			operatorKey, _ = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+			rootChainURL   = "ws://localhost:8546"
```

Comment:
use `cfg.RootChainURL ` instaed of  `rootChainURL`

Reply:
https://github.com/Onther-Tech/plasma-evm/pull/34/commits/700b686bd0b217d26b3ef2d8f8fa833ddfe1cda9